datafusion/common/src/dfschema.rs

```diff
@@ -899,6 +899,7 @@ impl TryFrom<SchemaRef> for DFSchema {
             field_qualifiers: vec![None; field_count],
             functional_dependencies: FunctionalDependencies::empty(),
         };
+        dfschema.check_names()?;
```

Comment:
Context is that checking names here makes this consistent with the other (fallible) ways to create a DFSchema from a Schema